src/prif.f90

```diff
@@ -10,7 +10,11 @@ module prif
   public :: prif_init
   public :: prif_stop, prif_error_stop, prif_fail_image
   public :: prif_allocate_coarray, prif_allocate, prif_deallocate_coarray, prif_deallocate
-  public :: prif_put, prif_put_indirect, prif_get, prif_get_indirect
+  public :: prif_put, prif_put_indirect, prif_get, prif_get_indirect, prif_put_with_notify, prif_put_with_notify_indirect
+  public :: prif_put_indirect_with_notify, prif_put_indirect_with_notify_indirect
+  public :: prif_get_strided, prif_get_strided_indirect, prif_put_strided, prif_put_strided_indirect
+  public :: prif_put_strided_with_notify, prif_put_strided_with_notify_indirect
+  public :: prif_put_strided_indirect_with_notify, prif_put_strided_indirect_with_notify_indirect
```

Comment:
Please remind me why we need this ugly ocean of `public` identifier declarations that are decoupled from the procedure interface declarations. This duplication adds maintenance headache, is there not a better way to do this?

Could we not instead attach the `public` attribute to each of the `module subroutine` interface declarations below and delete all these lines? 

Based on my AI Chatbot, module procedures declared in an `interface` block (like we have below) are automatically public, even in a module with default `private` accessibility; so we _might_ be able to just delete these lines with no further changes?